static/js/containers/Payment_test.js

```diff
@@ -1,90 +1,186 @@
+    });
   };
 
-  it('sets a price', () => {
-    let payment = renderPayment();
-    payment.find('input[id="total"]').props().onChange({
-      target: {
-        value: "123"
-      }
+  it('sets a selected klass', () => {
+    let fakeKlasses = generateFakeKlasses(3);
+    klassesStub = fetchStub.withArgs(klassesUrl)
+      .returns(Promise.resolve(fakeKlasses));
+    store.dispatch(setSelectedKlassIndex(2));
+
+    return renderFullPaymentPage().then((wrapper) => {
+      assert.deepEqual(wrapper.find('Payment').prop('selectedKlass'), fakeKlasses[2]);
     });
-    assert.equal(store.getState().ui.total, "123");
   });
 
-  it('sends a payment when API is contacted', () => {
-    store.dispatch(setTotal("123"));
-    sandbox.stub(api, 'fetchJSONWithCSRF').withArgs('/api/v0/payment/').returns(Promise.resolve());
-    let payment = renderPayment();
+  it('sets the selected klass to the first in the list by default', () => {
+    let fakeKlasses = generateFakeKlasses(3);
+    klassesStub = fetchStub.withArgs(klassesUrl)
+      .returns(Promise.resolve(fakeKlasses));
 
-    return listenForActions([REQUEST_PAYMENT, RECEIVE_PAYMENT_SUCCESS], () => {
-      payment.find('.payment-button').simulate('click');
+    return renderFullPaymentPage().then((wrapper) => {
+      assert.deepEqual(wrapper.find('Payment').prop('selectedKlass'), fakeKlasses[0]);
     });
   });
 
-  it('constructs a form to be sent to Cybersource and submits it', () => {
-    let url = '/x/y/z';
-    let payload = {
-      'pay': 'load'
-    };
-    sandbox.stub(api, 'fetchJSONWithCSRF').withArgs('/api/v0/payment/').returns(Promise.resolve({
-      'url': url,
-      'payload': payload
-    }));
-
-    let submitStub = sandbox.stub();
-    let fakeForm = document.createElement("form");
-    fakeForm.setAttribute("class", "fake-form");
-    fakeForm.submit = submitStub;
-    let createFormStub = sandbox.stub(util, 'createForm').returns(fakeForm);
-
-    let wrapper = renderPayment();
-    return listenForActions([REQUEST_PAYMENT, RECEIVE_PAYMENT_SUCCESS], () => {
-      wrapper.find('.payment-button').simulate('click');
-    }).then(() => {
-      return new Promise(resolve => {
-        setTimeout(() => {
-          assert.equal(createFormStub.callCount, 1);
-          assert.deepEqual(createFormStub.args[0], [url, payload]);
+  describe('UI', () => {
+    const klassTitleSelector = 'h2.klass-title',
+      klassDropdownSelector = 'select.klass-select';
+
+    it('shows the selected klass', () => {
+      let fakeKlasses = generateFakeKlasses(3);
+      klassesStub = fetchStub.withArgs(klassesUrl)
+        .returns(Promise.resolve(fakeKlasses));
+
+      return renderFullPaymentPage().then((wrapper) => {
+        let title = wrapper.find(klassTitleSelector);
+        assert.equal(title.text(), fakeKlasses[0].klass_name);
+      });
+    });
+
+    it('shows a dropdown if multiple klasses are available', () => {
+      for (let numKlasses of [1, 3]) {
```

Comment:
this may be personal preference, but for something like this (a sort of data-driven test) I prefer to fully parameterize it, so  something like:

```js
[
  [1, false],
  [2, true],
].forEach((numKlasses, shouldShowDropdown) => {
  // body of test
});
```

instead of computing `shouldShowDropdown` inside the test. imo it's more readable, up to you though!